app/scripts/minmax/dimMinMax.controller.js

```diff
@@ -169,27 +177,131 @@
+        vm.setOrderValues = vm.setOrder.split(',')
+      },
+      onDrop: function(dropped_id, type) {
+          dropped_id = dropped_id.split('-')[1];
+          var item = _.findWhere(buckets[vm.active][type], {id: dropped_id});
+          vm.lockeditems[type] = item;
+          var bestarmor = getBestArmor(buckets[vm.active], vm.lockeditems); 
+          vm.highestsets = vm.getSetBucketsStep(vm.active, bestarmor);
+      },
+      onRemove: function(removed_type) {
+          vm.lockeditems[removed_type] = null;
+          var bestarmor = getBestArmor(buckets[vm.active], vm.lockeditems); 
+          vm.highestsets = vm.getSetBucketsStep(vm.active, bestarmor);
+      },
+      getSetBucketsStep: function(activeGaurdian, bestArmor) {
+            var helms = bestArmor['helmet'];
+            var gaunts = bestArmor['gauntlets'];
+            var chests = bestArmor['chest'];
+            var legs = bestArmor['leg'];
+            var classItems = bestArmor['classItem'];
+            var ghosts = bestArmor['ghost'];
+            var artifacts = bestArmor['artifact'];
+            
+            var set_map = {};
+            var load_stats = function(set, hash, target_type) {
+                var total = 0;
+                _.each(set.armor, function(armor) {
+                    var stat = _.findWhere(armor.item.normalStats, {statHash: hash}) || {scaled: 0, bonus: 0};
+                    total += stat.scaled + (armor.bonus_type == target_type ? stat.bonus : 0);
+                });
+                return total;
+            };
+            
+            function step(activeGaurdian, h, g, c, l, ci, gh, ar, processed_count) {
+                for(; h < helms.length; ++h) {
+                for(; g < gaunts.length; ++g) {
+                for(; c < chests.length; ++c) {
+                for(; l < legs.length; ++l) {
+                for(; ci < classItems.length; ++ci) {
+                for(; gh < ghosts.length; ++gh) {
+                for(; ar < artifacts.length; ++ar) {
+                    var armor = {helmet: helms[h], gauntlets: gaunts[g], chest: chests[c], leg: legs[l], classItem: classItems[ci], ghost: ghosts[gh], artifact: artifacts[ar]};
+                    if(validSet(armor)) {
+                        var set = {armor: armor};
+                        set.int_val = load_stats(set, 144602215, 'int');
+                        set.disc_val = load_stats(set, 1735777505, 'disc');
+                        set.str_val = load_stats(set, 4244567218, 'str');
+                        var int_level = Math.min(Math.floor(set.int_val/60), 5);
+                        var disc_level = Math.min(Math.floor(set.disc_val/60), 5);
+                        var str_level = Math.min(Math.floor(set.str_val/60), 5);
+                        var tiers_string = int_level.toString() + '/' + disc_level.toString() + '/' + str_level.toString();
+                        if(tiers_string in set_map) {
+                            set_map[tiers_string].push(set);
```

Comment:
might I suggest adding some logic here that eliminates identical sets (same tier, and all the same identical items.) There have been cases where I can hit 5/5/2 with the same exact armors, but for example instead using the int bonus on my class item and the dis bonus on my ghost, right now you'll see another option that is to use the dis bonus on the class item, and the int bonus on the ghost... when in the end, it's the exact same.

Something like this:
```
                            var idxmap = _.map(set.armor, function(t) { return t.item.index; });
                            var exists = _.find(set_map[tiers_string], function(s) {
                                return _.isEqual(_.map(s.armor, function(t) { return t.item.index; }), idxmap);
                            });
                            if(!exists) {
                                set_map[tiers_string].push(set);
                            }
```